flink-connectors/flink-connector-rabbitmq2/src/test/java/org/apache/flink/connector/rabbitmq2/source/RabbitMQSourceITCase.java

```diff
@@ -0,0 +1,227 @@
+        DataStream<String> stream = addSourceOn(env, ConsistencyMode.AT_LEAST_ONCE);
+
+        List<String> messages = getSequentialMessages(100);
+        failAtNthMessage = 30;
+        messageLatch = new CountDownLatch(messages.size() + failAtNthMessage - 1);
+        stream.addSink(new CollectSink());
+
+        env.executeAsync();
+
+        sendToRabbit(messages);
+        messageLatch.await();
+
+        assertTrue(collectedMessages.containsAll(messages));
+    }
+
+    // --------------- exactly once ---------------
+    @Test
+    public void exactlyOnceTest() throws Exception {
+        List<String> messages = getRandomMessages(1000);
+        messageLatch = new CountDownLatch(messages.size());
+
+        DataStream<String> stream = addSourceOn(env, ConsistencyMode.EXACTLY_ONCE);
+        stream.addSink(new CollectSink());
+        env.executeAsync();
+
+        // use messages as correlation ids here
+        sendToRabbit(messages, messages);
+
+        messageLatch.await();
+
+        assertEquals(messages, collectedMessages);
+    }
+
+    @Test
+    public void exactlyOnceFilterCorrelationIdsTest() throws Exception {
+        List<String> messages = getRandomMessages(5);
+        messageLatch = new CountDownLatch(3);
+
+        env.enableCheckpointing(5000);
+        DataStream<String> stream = addSourceOn(env, ConsistencyMode.EXACTLY_ONCE);
+        stream.addSink(new CollectSink());
+        env.executeAsync();
+
+        List<String> correlationIds = Arrays.asList("1", "2", "3", "3", "3");
+        sendToRabbit(messages, correlationIds);
+
+        messageLatch.await();
+
+        List<String> expectedMessages = messages.subList(0, 3);
+        assertEquals(expectedMessages, collectedMessages);
+    }
+
+    @Test
+    public void exactlyOnceWithFailureAndMessageDuplicationTest() throws Exception {
+        // An exception is thrown in order to trigger a restart of Flink and it
+        // is assured that the system receives the messages only once. We disable
+        // (by setting the interval higher than the test duration) checkpoint to
+        // expect receiving all pre-exception messages once again.
+        env.enableCheckpointing(500000);
+        DataStream<String> stream = addSourceOn(env, ConsistencyMode.EXACTLY_ONCE);
+
+        List<String> messages = getRandomMessages(100);
+
+        int originalFailAthNthMessage = 30;
+        failAtNthMessage = originalFailAthNthMessage;
+        messageLatch = new CountDownLatch(messages.size() + failAtNthMessage - 1);
+        stream.addSink(new CollectSink());
+        env.executeAsync();
+
+        sendToRabbit(messages, messages);
+        messageLatch.await();
+
+        List<String> expectedMessage =
+                collectedMessages.subList(originalFailAthNthMessage - 1, collectedMessages.size());
+        assertEquals(messages, expectedMessage);
+    }
+
+    @Test
+    public void exactlyOnceWithFailureWithNoMessageDuplicationTest() throws Exception {
```

Comment:
I am not sure I understand the difference between both tests. Can you either rename one of them or add a small docstring want kind of scenario you are testing?